docs/content.zh/docs/deployment/resource-providers/native_kubernetes.md

```diff
@@ -324,4 +324,235 @@ $ kubectl create clusterrolebinding flink-role-binding-flink --clusterrole=edit
 
 Please refer to the official Kubernetes documentation on [RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) for more information.
 
+### Pod Template
+
+Flink allows users to define the JobManager and TaskManager pods via template files, with the advanced features (e.g. volumes, init containers, sidecar containers, etc.)
+that are not supported by Flink [Kubernetes config options]({{< ref "docs/deployment/config" >}}#kubernetes) directly.
+Use [`kubernetes.pod-template-file`]({{< ref "docs/deployment/config" >}}#kubernetes-pod-template-file)
+to specify a local file that contains pod definition. It will be used, instead of an empty pod, to initialize the
+JobManager and TaskManager. The main container should be defined with name `flink-main-container`.
+Please refer to the [pod template example](#example-of-pod-template) for more information.
+
+#### Fields Overwritten by Flink
+
+Some fields of the pod template can be overwritten by Flink.
+The overwritten mechanism can be categorized as follows.
+* **Defined by Flink.** Definitions in the template won't take effect. User cannot control the fields.
+* **Defined by Config Option.** Definitions in the template won't take effect. User can control the fields via config options.
+* **Overwritten by Flink.** Definitions in the template will take effect, with additional contents from Flink.
```

Comment:
Overwritten by Flink sounds as if we are replacing the template fields with something. Maybe "Merged with Config Option" describes it a bit better.